src/gennylib/src/PoolManager.cpp

```diff
@@ -55,6 +62,10 @@ mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
     // ...but no need to keep the lock open past this.
     // Two threads trying access client("foo",0) at the same
     // time will subsequently block on the unique_lock.
+
+    if (!_encryptionManager) {
+        _encryptionManager = std::make_unique<EncryptionManager>(context, _dryRun);
+    }
```

Comment:
(nit) I think the above comment about releasing the lock should be after this encryption manager call

Reply:
fixed